molgenis-data-platform/src/main/java/org/molgenis/data/platform/decorators/MolgenisRepositoryDecoratorFactory.java

```diff
@@ -80,23 +84,26 @@ public MolgenisRepositoryDecoratorFactory(EntityManager entityManager,
 	{
 		Repository<Entity> decoratedRepository = repository;
 
-		// 14. Query the L2 cache before querying the database
+		// 15. Query the L2 cache before querying the database
 		decoratedRepository = new L2CacheRepositoryDecorator(decoratedRepository, l2Cache, transactionInformation);
 
-		// 13. Query the L1 cache before querying the database
+		// 14. Query the L1 cache before querying the database
 		decoratedRepository = new L1CacheRepositoryDecorator(decoratedRepository, l1Cache);
 
-		// 12. Route specific queries to the index
+		// 13. Route specific queries to the index
 		decoratedRepository = indexedRepositoryDecoratorFactory.create(decoratedRepository);
 
-		// 11. Query the L3 cache before querying the index
+		// 12. Query the L3 cache before querying the index
 		decoratedRepository = new L3CacheRepositoryDecorator(decoratedRepository, l3Cache, transactionInformation);
 
-		// 10. Register the cud action needed to index indexed repositories
+		// 11. Register the cud action needed to index indexed repositories
 		decoratedRepository = new IndexActionRepositoryDecorator(decoratedRepository, indexActionRegisterService);
 
-		// 9. Custom decorators
-		decoratedRepository = repositoryDecoratorRegistry.decorate(decoratedRepository);
+		// 10. Dynamic decorators
+		decoratedRepository = dynamicRepositoryDecoratorRegistry.decorate(decoratedRepository);
```

Comment:
Security: as a user can a decorate a repository that I don't have permission to read and e.g. email all traffic to my mail address?

Reply:
In theory: yes, but that would imply two things:
1. The admin gave a user permission to edit the configuration entity, which he/she shouldn't do because it's a high-level admin feature.
2. The evil user managed to add a malicious decorator to the code base and deployed the new artifact to the server without the admins credentials.

Reply:
It gets trickier once we add a dynamic decorator that executes scripts that can be modified by the user (side note: script executing decorator, sounds like ultimate flexibility). Why not execute your decorator right after the repository call to prevent any security issues?

Reply:
That would make decorators that change values impossible I think?

Letting users edit scripts that are triggered with system permissions by a decorator is something you shouldn't do if you ask me. But yes, it is possible to write 'dangerous' decorators and admins should know what they are doing.

Reply:
> That would make decorators that change values impossible I think?

Why? You should have permissions to change the values of course.

> Letting users edit scripts that are triggered with system permissions by a decorator is something you shouldn't do if you ask me

Too dynamic? ;-D